airbyte-integrations/connectors/source-monday/source_monday/item_pagination_strategy.py

```diff
@@ -45,3 +49,52 @@ def next_page_token(self, response, last_records: List[Mapping[str, Any]]) -> Op
                 return None
 
         return self._page, self._sub_page
+
+
+class ItemCursorPaginationStrategy(PageIncrement):
+    """
+    Page increment strategy with subpages for the `items` stream.
+
+    From the `items` documentation https://developer.monday.com/api-reference/docs/items:
+        Please note that you cannot return more than 100 items per query when using items at the root.
+        To adjust your query, try only returning items on a specific board, nesting items inside a boards query,
+        looping through the boards on your account, or querying less than 100 items at a time.
+
+    This pagination strategy supports nested loop through `boards` on the top level and `items` on the second.
+    See boards documentation for more details: https://developer.monday.com/api-reference/docs/boards#queries.
+    """
+
+    def __post_init__(self, parameters: Mapping[str, Any]):
+        # `self._page` corresponds to board page number
+        # `self._sub_page` corresponds to item page number within its board
+        self.start_from_page = 1
+        self._page: Optional[int] = self.start_from_page
+        self._sub_page: Optional[int] = self.start_from_page
+
+    def next_page_token(self, response, last_records: List[Mapping[str, Any]]) -> Optional[Tuple[Optional[int], Optional[int]]]:
+        """
+        `items` stream use a separate 2 level pagination strategy where:
+        1st level `boards` - incremental pagination
+        2nd level `items_page` - cursor pagination
+
+        Attributes:
+            response: Contains `boards` and corresponding lists of `items` for each `board`
+            last_records: Parsed `items` from the response
+        """
+        data = response.json()["data"]
+        boards = data.get("boards", [])
+        next_items_page = data.get("next_items_page", {})
+        if boards:
+            board = boards[0]
```

Comment:
Can we clarify why we can just take the first one? My hypothesis based on [that](https://github.com/airbytehq/airbyte/blob/ea07171669381f355059720a11a2889c8f65f00b/airbyte-integrations/connectors/source-monday/source_monday/manifest.yaml#L182) is that we only request one board at the same time but it is very unclear for this context. Even the documentation is confusing as it states `This pagination strategy supports nested loop through 'boards' on the top level` but there is not really a capability to loop over multiple boards

Reply:
there is always only one board due to limit=1, so in one request we extract all 'items_page' for one board only. 
We actually loop over boards: we 1 board per page.